Signum.Engine.Extensions/Mailing/EmailLogic.cs

```diff
@@ -314,6 +314,7 @@ public static void Register()
 
 public interface IEmailSenderManager
 {
+    Func<EmailTemplateEntity?, Lite<Entity>?, EmailMessageEntity?, EmailSenderConfigurationEntity> GetEmailSenderConfiguration { get; }
     void Send(EmailMessageEntity email);
 }
 

```

Comment:
I don't understand why you need to have the delegate inside the IEmailSenderManager. 

Before this change, the IEmailSenderManager was a global extension point to control how to send emails. 

Now I think it makes sense to have a Polymorphic<Func<EmailSenderServiceConfigurationEntity, IEmailSenderManager>> and separate the implementations that now are in partial classes like [EmailSenderManager.Exchange.cs, EmailSenderManager.MicrosoftGraph.cs, etc.. Into separate classes. 

Maybe also rename IEmailSenderManager -> IEmailSender and then have in the polymorphic the following mapping: 

`ExchangeWebServiceSenderEntity` -> `ExchangeWebServiceSender`, 
`MicrosoftGrapSenderhEntity` -> `MicrosoftGraphSender`
`SmtpSenderEntity` -> `SmptSender`
`YourServiceSenderEntity` -> `YourServiceSender`